hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarbonitingMetadataFreezerActor.scala

```diff
@@ -0,0 +1,124 @@
+package cromwell.services.metadata.hybridcarbonite
+
+import java.nio.file.StandardOpenOption
+
+import akka.actor.{ActorRef, FSM, LoggingFSM}
+import cats.data.NonEmptyList
+import cromwell.core.WorkflowId
+import cromwell.core.io.{AsyncIo, DefaultIoCommandBuilder}
+import cromwell.services.MetadataServicesStore
+import cromwell.services.metadata.MetadataArchiveStatus.{ArchiveFailed, Archived}
+import cromwell.services.metadata.MetadataService.GetMetadataAction
+import cromwell.services.metadata.hybridcarbonite.CarbonitingMetadataFreezerActor._
+import cromwell.services.metadata.impl.MetadataDatabaseAccess
+import cromwell.services.metadata.impl.builder.MetadataBuilderActor.{BuiltMetadataResponse, FailedMetadataResponse}
+import cromwell.services.metadata.{MetadataArchiveStatus, MetadataQuery}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+final class CarbonitingMetadataFreezerActor(carboniterConfig: HybridCarboniteConfig,
+                                            serviceRegistry: ActorRef,
+                                            ioActor: ActorRef) extends
+  LoggingFSM[CarbonitingMetadataFreezingState, CarbonitingMetadataFreezingData]
+  with MetadataDatabaseAccess
+  with MetadataServicesStore {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  val asyncIo = new AsyncIo(ioActor, DefaultIoCommandBuilder)
+
+  startWith(Pending, NoData)
+
+  when(Pending) {
+    case Event(FreezeMetadata(workflowId), NoData) =>
+      val getMetadataAction = GetMetadataAction(MetadataQuery(
+        workflowId = workflowId,
+        jobKey = None,
+        key = None,
+        includeKeysOption = None,
+        excludeKeysOption = Option(NonEmptyList.of("labels")),
+        expandSubWorkflows = true
+      ))
+
+      serviceRegistry ! getMetadataAction
+      goto(Fetching) using FetchingData(workflowId)
+  }
+
+  when(Fetching) {
+    case Event(BuiltMetadataResponse(_, responseJson), FetchingData(workflowId)) =>
+      asyncIo.writeAsync(carboniterConfig.makePath(workflowId), responseJson.compactPrint, Seq(StandardOpenOption.CREATE)) onComplete {
```

Comment:
compact or pretty? yes the former takes less space but the latter would make ad hoc inspection easier

Reply:
Added to tomorrow's tech talk